src/main/kotlin/uk/gov/dluhc/printapi/mapper/IssuerDtoMapper.kt

```diff
@@ -0,0 +1,28 @@
+package uk.gov.dluhc.printapi.mapper
+
+import org.mapstruct.Mapper
+import org.mapstruct.Mapping
+import uk.gov.dluhc.eromanagementapi.models.ContactDetails
+import uk.gov.dluhc.eromanagementapi.models.LocalAuthorityResponse
+import uk.gov.dluhc.printapi.dto.IssuerContactDetailsDto
+import uk.gov.dluhc.printapi.dto.IssuerDto
+
+@Mapper
+abstract class IssuerDtoMapper {
+
+    @Mapping(target = "englishContactDetails", expression = "java(toIssuerContactDetailsDto( localAuthority.getName(), localAuthority.getContactDetailsEnglish() ))")
+    @Mapping(target = "welshContactDetails", expression = "java(toNullIssuerContactDetailsDto( localAuthority.getName(), localAuthority.getContactDetailsWelsh() ))")
+    abstract fun toIssuerDto(localAuthority: LocalAuthorityResponse): IssuerDto
+
+    fun toNullIssuerContactDetailsDto(name: String, contactDetails: ContactDetails?): IssuerContactDetailsDto? {
```

Comment:
We have a bit of a mismatch between what data is available and what data we need - this block only returns a complete contact details. (Otherwise a partial one is attempted using only the name). A future story is likely to return the welsh name and that will resolve this issue.